bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/DateTimeType.java

```diff
@@ -120,7 +121,7 @@ public static DateTimeType valueOf(String value) {
     }
 
     @Override
-    public String format(String pattern) {
+    public String format(@Nullable String pattern) {

```

Comment:
Is there a specific reason to add the `@Nullable` annotation here - we have it in `QuantityType` too - whereas the interface `Type` as well all other inherited classes use `@NonNullByDefault` from class level?

Reply:
There's no specific reason for this so I think it would make more sense to remove `@Nullable`'s in the overridden methods. Unless of course the format method should be null proof to prevent NPEs when they are used in scripting engines.

Reply:
IIRC the `String#format` method is used by all other `Type`s, too.